### ISSUE TITLE
3-D object captured in a 2-D representation

### ISSUE THREAD

**Issue by Timothyrichpt** [charlieforward9/visualPT#211]
The only way we can validate images is by the height of the Subject which will vary by the Subject.

Look at the height of my fingers on my shoulders, relative to the horizontal lines on the credenza in the background. 

In several of the figures, which are vertical,my fingers are much lower, indicating backward sway. 

![image](https://github.com/VisualPT/visualPT/assets/124405841/d0969a21-53d0-4b61-a9e6-e94e28c22839)

![image](https://github.com/VisualPT/visualPT/assets/124405841/82f63c72-6785-4e50-b8ae-63018fbc7e37)

![image](https://github.com/VisualPT/visualPT/assets/124405841/1291ca34-6571-4e50-8e21-fcc7f9224521)

![image](https://github.com/VisualPT/visualPT/assets/124405841/b3f3c7fb-ff5a-4976-b1a1-f476f911b8e9)

**Comment by charlieforward9** [comment]
@Timothyrichpt what is the action item?

**Comment by Timothyrichpt** [comment]
3 ideas to capture the 3rd dimension:
1. @DrBubbles42 would use math to capture the Z-axis
2. we could use 2 cameras/mobile devices with the app
3. pivot to a hardware solution with a 3D camera for the computer vision application

@charlieforward9

**Comment by charlieforward9** [comment]
The MediaPipe Pose Detection model has a z-index output described in [this document](https://drive.google.com/file/d/10WlcTvrQnR_R2TdTmKw0nkyRLqrwNkWU/preview):

> Z coordinate is measured in "image pixels" like
the X and Y coordinates and represents the
distance relative to the plane of the subject's
hips, which is the origin of the Z axis. Negative
values are between the hips and the camera;
positive values are behind the hips. Z coordinate
scale is similar with X, Y scales but has different
nature as obtained not via human annotation, by
fitting synthetic data (GHUM model) to the 2D
annotation. Note, that Z is not metric but up to
scale.

This is currently being captured by Nick. However, he said it is not very consistent. 

@DrBubbles42 could you share some samples of individual outputs on 3-axis to see how the z-index affects the output. (3D graphs of the individual wireframes)

**Comment by DrBubbles42** [comment]
Yes I can provide some individual samples later today or tomorrow at the latest.

**Comment by Timothyrichpt** [comment]
The first and the third image look very promising!

**Comment by DrBubbles42** [comment]
I have made a mistake in the output i will delete the post and fix it

**Comment by DrBubbles42** [comment]
Here are a couple of snips of the 3d data. The first image is from the front(if a negative z denotes closer to the camera) and the next on is from the (left) side to show the z values. All of the green points are no sway (success) and the reds are sway (fail). If you want to see and explore for yourself we can hop on a call some time and i can show you how to run it.
![image](https://github.com/VisualPT/visualPT/assets/25064175/bba94c16-ec02-49ba-96e3-e9c5dd206265)
![image](https://github.com/VisualPT/visualPT/assets/25064175/cb8adebb-5323-4a58-a40c-e171fc85514b)
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------
![image](https://github.com/VisualPT/visualPT/assets/25064175/c3e7a0b1-6261-44aa-8a20-52379e8a3e6f)
![image](https://github.com/VisualPT/visualPT/assets/25064175/3b41e72e-a503-4c1b-906f-ebd112f8c0a5)
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------
![image](https://github.com/VisualPT/visualPT/assets/25064175/37f3165d-2b9f-4b99-9ae9-538e845b7321)
![image](https://github.com/VisualPT/visualPT/assets/25064175/243b3f60-7c6c-464e-83c9-4ce17d0f5b3d)
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------
![image](https://github.com/VisualPT/visualPT/assets/25064175/9c308c6e-6614-4bb8-bb83-2b514be9f82d)
![image](https://github.com/VisualPT/visualPT/assets/25064175/76d7b02a-797a-4046-94e9-a9091c84c957)

As seen from the images the z values are stretched out a bit. This also highlights my issues that Im having with normalizing the data to get it all centered. The last success point is tough to see how a person would stand in that position. some of the data points are clumped together making a skinny figure and some are more spread out making a fat figure.

**Comment by charlieforward9** [comment]
Transferring this to the balance model repo as I do not think there is not much else I can do about it within the app.

**Comment by charlieforward9** [comment]
Conversation about integrating LiDAR initated [here](https://github.com/olexale/arkit_flutter_plugin/discussions/234)

**Comment by charlieforward9** [comment]
 technical research on their Depth Pro technology https://github.com/apple/ml-depth-pro

**Comment by charlieforward9** [comment]
<img width="1351" alt="Image" src="https://github.com/user-attachments/assets/1dbae624-b2b0-405a-9297-2bffe599f95d" />

Ref: https://www.reddit.com/r/computervision/comments/1gdsm0b/best_depth_estimation_model_depth_anything_v2/

**Comment by Timothyrichpt** [comment]
> <img alt="Image" width="1351" src="https://private-user-images.githubusercontent.com/62311337/399181384-1dbae624-b2b0-405a-9297-2bffe599f95d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU1MDA2ODUsIm5iZiI6MTczNTUwMDM4NSwicGF0aCI6Ii82MjMxMTMzNy8zOTkxODEzODQtMWRiYWU2MjQtYjJiMC00MDVhLTkyOTctMmJmZmU1OTlmOTVkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjI5VDE5MjYyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFhNmY2NTU1MWZiNTFhYjRiMTVlNGFiMGE4ZTI0MDk5NzY1ZTQ1MmE5OWQzMzMyZTllZTQwMGNjMTc0NGU2ZTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.28FgK_ArbYXsUm8H69ye1VqnTniK3GxoBIdva5Oh4pM">
> Ref: https://www.reddit.com/r/computervision/comments/1gdsm0b/best_depth_estimation_model_depth_anything_v2/

@charlieforward9 DepthPro supports video but it also has the largest file size (harder to work with?) and requires a TenCent license (China?)

**Comment by charlieforward9** [comment]
Yeah I wasn't aware of this, glad I watched this video before going down that rabbit hole.

I am going to look at Metric3D instead. I have [requested mobile support](https://github.com/YvanYin/Metric3D/issues/193)

**Comment by charlieforward9** [comment]
@Timothyrichpt i need you to set up your  Developer account to be a part of the Founders team so i can send you alpha stage app releases and you can test this with people.

**Comment by charlieforward9** [comment]
This issue is largely delegated to the [KinesteX team](https://github.com/KinesteX). I will close this one as a duplicate and keep the parent issue around for a bit longer.
